corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -45,11 +48,38 @@ hqDefine("data_dictionary/js/data_dictionary", [
                     propObj.deprecated.subscribe(changeSaveButton);
                     propObj.removeFHIRResourcePropertyPath.subscribe(changeSaveButton);
                     propObj.allowedValues.on('change', changeSaveButton);
-                    self.properties.push(propObj);
-                });
-            });
+                    groupObj.properties.push(propObj);
+                }
+                groupObj.properties.subscribe(changeSaveButton);
+                self.groups.push(groupObj);
+            }
+        };
+
+        return self;
+    };
+
+    var groupsViewModel = function (id, name, description, caseType, deprecated) {
+        var self = {};
+        self.id = id;
+        self.name = ko.observable(name);
+        self.description = ko.observable(description);
+        self.caseType = caseType;
+        self.properties = ko.observableArray();
+        self.expanded = ko.observable(true);
+        self.toggleExpanded = () => self.expanded(!self.expanded());
+        self.deprecated = deprecated;
+        // Ensures that groups are not directly hidden on clicking the deprecated button
+        self.toBeDeprecated = ko.observable(deprecated || false);
+        self.deprecateGroup = function () {
+            self.toBeDeprecated(true);
         };
 
+        self.restoreGroup = function () {
+            self.toBeDeprecated(false);
+        };
+        self.showDeprecatedWarning = ko.computed(function () {
+            return self.toBeDeprecated() && !deprecated && self.properties().length > 0;
```

Comment:
This is used to show the warning that the properties will be moved to "No Group" section, so that checks the groups with no properties as they would not need that warning.

Reply:
Alright,
Can we use better naming here then? I got confused because to deprecate a group, it should not be needed that it has properties.
Like `showNoGroupTransferMessage` or something better.

Reply:
Fixed. [3b0ca1f](https://github.com/dimagi/commcare-hq/pull/32878/commits/3b0ca1f0df86c95684cacac9c21795c30cbcf030)